lib/src/folding/parsers/indent.dart

```diff
@@ -0,0 +1,96 @@
+import 'dart:collection';
+
+import 'package:highlight/highlight.dart';
+
+import '../../code/code_line.dart';
+import '../foldable_block.dart';
+import '../foldable_block_type.dart';
+import 'abstract.dart';
+
+/// A parser for foldable blocks from lines indentation
+class IndentFoldableBlockParser extends AbstractFoldableBlockParser {
+  static const _kSeparatorLine = -1;
+
+  final openBlocksLinesByIndent = HashMap<int, List<int>>();
+
+  @override
+  void parse(
+    Result highlighted,
+    Set<Object?> serviceCommentsSources,
+    List<CodeLine> lines,
+  ) {
+    _addWhitespacesBlocks(lines);
+    finalize();
+  }
+
+  void _addWhitespacesBlocks(List<CodeLine> lines) {
+    final lineIndents = _calculateLineIndents(lines);
+
+    int lastValuableLine = lineIndents.length - 1;
```

Comment:
renamed to lastNonEmptyLine. This variable helps parser to recognize case, where there are several separator lines at the end of code. It will reassign in the loop in this case.

Reply:
I'll keep this open for now and maybe deal with this later. If we are keeping this algo for now, this will also do.